chainreg/chainregistry.go

```diff
@@ -448,6 +452,8 @@ func NewPartialChainControl(cfg *Config) (*PartialChainControl, func(), error) {
 		)
 
 		cc.ChainNotifier = chainNotifier
+		cc.CurrentChainStateTracker =
+			chainntnfs.NewChainStateTracker(&cc.ChainNotifier)
```

Comment:
Ah ok I see here when it's create, notifier isn't fully spun up at this point, so can't just pass the epoch in. Could also go a route of just defining a sub-interface with that single method, so then reduces concerns when it comes to testing/mocking.

Reply:
Which "single method" are you referring to? I generally try to avoid interfaces with a single method as it's hard to attest to correct functioning of an interface when it only has one method. Most invariants are expressed by relationships between methods. Idempotence is a notable exception to this.